plenum/server/replica.py

```diff
@@ -1688,6 +1706,87 @@ def can_pp_seq_no_be_in_view(self, view_no, pp_seq_no):
                                           compare_3PC_keys((view_no, pp_seq_no),
                                                            self.last_prepared_before_view_change) >= 0)
 
+    def request_pre_prepare_if_possible(self, three_pc_key) -> bool:
```

Comment:
Please ass tests (probably even unit tests) to check each of the conditions below.